src/sliding_sync.rs

```diff
@@ -1179,7 +1202,8 @@ pub fn block_on_async_with_timeout<T>(
     timeout: Option<Duration>,
     async_future: impl Future<Output = T>,
 ) -> Result<T, Elapsed> {
-    let rt = TOKIO_RUNTIME.get_or_init(|| Arc::new(tokio::runtime::Runtime::new().unwrap()));
+    let mut binding = TOKIO_RUNTIME.lock().unwrap();
+    let rt = binding.get_or_insert_with(|| Arc::new(tokio::runtime::Runtime::new().unwrap()));
```

Comment:
if you create a separate binding, the lock surrounding the tokio runtime is held until the end of this scope. That isn't necessary, so please remove the binding and combine them like so:

```suggestion
    let rt = TOKIO_RUNTIME.lock().unwrap().get_or_insert_with(||
        Arc::new(tokio::runtime::Runtime::new().unwrap())
    );
```

Reply:
use separate binding is a bad pratice and I just change it